addons/mail/controllers/google_translate.py

```diff
@@ -0,0 +1,51 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route, Controller
+
+import babel
+import requests
+
+
+class GoogleTranslateController(Controller):
+    @route("/mail/message/translate", type="json", auth="user")
+    def translate(self, message_id):
+        message = request.env["mail.message"].browse(message_id)
+        message._validate_access_for_current_persona("read")
```

Comment:
Isn't the method returning a boolean that needs to be checked and manually raise in case it's false?

Reply:
The returned boolean indicates that both "check_access_rule" and "check_access_rights" were called on a non empty recordset. The method do raise an error if needed and this test assert it: https://github.com/odoo/odoo/pull/126555/files#diff-7256d5a8766d092d6ee713f31f8d79bb6187bd0054ec0b87b427c0cf21cd3911R134

Reply:
Ok but then it's still good to check the return value and raise if empty to prevent from executing the rest of the code on a non-existing message.